gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedExtractor.java

```diff
@@ -67,12 +75,19 @@ public FileBasedExtractor(WorkUnitState workUnitState, FileBasedHelper fsHelper)
         Lists.newArrayList(workUnitState.getPropAsList(ConfigurationKeys.SOURCE_FILEBASED_FILES_TO_PULL, ""));
     this.statusCount = this.workUnit.getPropAsInt(ConfigurationKeys.FILEBASED_REPORT_STATUS_ON_COUNT,
         ConfigurationKeys.DEFAULT_FILEBASED_REPORT_STATUS_ON_COUNT);
-    this.fsHelper = fsHelper;
+
+    if (fsHelper instanceof SizeAwareFileBasedHelper) {
+      this.fsHelper = (SizeAwareFileBasedHelper)fsHelper;
+    } else {
+      this.fsHelper = new SizeAwareFileBasedHelperDecorator(fsHelper);
```

Comment:
fsHelper is guaranteed to be a FileBasedHelper as it is passed in as a FileBasedHelper in the constructor (line 71)